tests/core/io/test_resource_uid.h

```diff
@@ -47,6 +47,27 @@ TEST_CASE("[ResourceUID] Must encode/decode maximum/minimum UID correctly") {
 	CHECK_MESSAGE(ResourceUID::get_singleton()->text_to_id("uid://a") == 0, "Minimum UID must decode correctly.");
 }
 
+TEST_CASE("[ResourceUID] Must encode and decode invalid UIDs correctly") {
+	ResourceUID *rid = ResourceUID::get_singleton();
+	CHECK_MESSAGE(rid->id_to_text(-1) == "uid://<invalid>", "Invalid UID -1 must encode correctly.");
+	CHECK_MESSAGE(rid->text_to_id("uid://<invalid>") == -1, "Invalid UID -1 must decode correctly.");
+
+	CHECK_MESSAGE(rid->id_to_text(-2) == rid->id_to_text(-1), "Invalid UID -2 must encode to the same as -1.");
+}
+
+TEST_CASE("[ResourceUID] Must encode and decode various UIDs correctly") {
+	ResourceUID *rid = ResourceUID::get_singleton();
+	CHECK_MESSAGE(rid->id_to_text(1) == "uid://b", "UID 1 must encode correctly.");
+	CHECK_MESSAGE(rid->text_to_id("uid://b") == 1, "UID 1 must decode correctly.");
+
+	CHECK_MESSAGE(rid->id_to_text(8060368642360689600) == "uid://dm3rdgs30kfci", "A normal UID must encode correctly.");
+	CHECK_MESSAGE(rid->text_to_id("uid://dm3rdgs30kfci") == 8060368642360689600, "A normal UID must decode correctly.");
+
+	// Note that a UID with a leading 'a' is treated like a leading zero.
+	CHECK_MESSAGE(rid->id_to_text(36) == "uid://bc", "UID with 'a' (zero) must encode correctly.");
+	CHECK_MESSAGE(rid->text_to_id("uid://abc") == 36, "UID with 'a' (zero) must decode correctly.");
```

Comment:
That looks unexpected. Editor shouldn't be able to serialize UIDs that deserialize differently, so I'm not sure if it's worth testing them. If anything, I'd test that such UID can't be generated normally.

Reply:
Removed :)